proxy/http/parser/parser.py

```diff
@@ -300,17 +299,28 @@ def _process_line_and_headers(self, raw: bytes) -> Tuple[bool, bytes]:
         return len(raw) > 0, raw
 
     def _process_line(self, raw: bytes) -> None:
+        # defer import to solve the circular import problem
+        from ..exception import HttpProtocolException
```

Comment:
For now let's just throw 'NotImplementedError` and get past this PR.  We can always clean-up in a follow up PR, because we are running into circular import exception.  If fixing circular import is not straightforward, let's not try to fight two things in a single PR.  Leave a `TODO: Runs into circular import error when using HttpProtocolException` and possibly link to this comment.  Thank you!!!

Reply:
Got it.  Have updated the codes accordingly, thank you